agent/agent.go

```diff
@@ -1246,8 +1246,12 @@ func (a *Agent) killProcs(ctx context.Context, tc *taskContext, ignoreTaskGroupC
 	}
 }
 
-func (a *Agent) clearGitConfig(tc *taskContext, reason string) {
-	logger := grip.NewJournaler("clearGitConfig")
+func (a *Agent) clearGitConfig(tc *taskContext, ignoreTaskGroupCheck bool, reason string) {
```

Comment:
Rather than passing a true/false bool to ignore the task group check, if we're letting the gitconfig persist for the entirety of the task group and clearing it after every task, then it pretty much behaves the same as task directory cleanup. In that case, it should be sufficient to just run `clearGitConfig` in a defer in `runTeardownGroupCommands`, like [this](https://github.com/evergreen-ci/evergreen/blob/79237aadc82560a18615f4212fa5ba68d71bf7c8/agent/agent.go#L869). That line cleans up the task directory for both the end of a regular task and at the end of task groups.